src/writer.rs

```diff
@@ -143,6 +143,24 @@ impl Writer<Vec<u8>> {
             Ok(()) => &**self.buf.get_ref(),
         }
     }
+
+    /// Returns the written CSV data as owned string.
+    pub fn to_string(self) -> String {
```

Comment:
Could you rename this to `into_string`? (Per conventions in `std`.) Similarly for `to_bytes` -> `into_bytes`.